btree/btree.c

```diff
@@ -3415,11 +3448,23 @@ int64_t btree_create_tick(struct m0_sm_op *smop)
 		oi->i_nop.no_tree->t_type = data->bt;
 		node_init(&oi->i_nop, k_size, v_size, bop->bo_tx);
 
-		m0_rwlock_write_lock(&bop->bo_arbor->t_lock);
 		bop->bo_arbor->t_desc           = oi->i_nop.no_tree;
 		bop->bo_arbor->t_type           = data->bt;
+
+		m0_rwlock_write_lock(&bop->bo_arbor->t_desc->t_lock);
 		bop->bo_arbor->t_desc->t_height = 1;
-		m0_rwlock_write_unlock(&bop->bo_arbor->t_lock);
+		/** Initialtise active nd list for this tree. */
+		ndlist_tlist_init(&bop->bo_arbor->t_desc->t_active_nds);
+		/**
+		 * Deliberately increasing the ref count of root node, so that
+		 * it is never freed (except for in btree_destroy). Add root
+		 * node to active nd list.
+		 */
+		bop->bo_arbor->t_desc->t_root->n_ref++;
```

Comment:
The internals of the node should not be exposed here. Some interface needs to be defined. It can be done later though just make a note of it someplace.